resolwe_bio/processes/clustering/hierarchical_clustering.yml

```diff
@@ -123,21 +123,41 @@
     runtime: polyglot
     language: bash
     program: |
-      {% if genes %}
-        {% if exps|selectattr('source', 'equalto', source)|list|length != exps|length %}
-          re-error "Source of selected genes must be the same as the source of all expression files."
+
+      {% for e in exps %}
```

Comment:
@mzganec Please advise if @JenkoB could use `|all` filter here.

Reply:
As discussed with @mzganec ``|all filter`` can not be used here, since we are searching for exact sample which caused Error, so we can not  avoid ``for loops``.  I have also removed all``or`` conditions, because they are not needed.